### PR TITLE
Implements handler for special services activation

### PR DESCRIPTION
**Pull request by CrisTofani**
This PR aims to properly handle the special services activation status. 
CGN activation now effects the related service activation

### CODE REVIEW

File: src/payloads/services/index.ts [hunk 1]
```diff
@@ -74,11 +78,39 @@ export const getServicesTuple = (
...
+>([["cgn", isCgnActivated]]);
+
 export const getServicesPreferences = (
   services: ReadonlyArray<ServicePublic>
 ) =>
   new Map<ServiceId, ServicePreference>(
     services.map(s => {
+      const metadata = s.service_metadata;
+      if (metadata && SpecialServiceMetadata.is(metadata)) {
+        const specialServiceInbox = fromNullable(metadata.custom_special_flow)
```
Comment (pietro909) [inline src/payloads/services/index.ts:93]: It seems like `specialServiceInbox` is a boolean: maybe `hasSpecialServiceInbox` would better convey this semantics?

File: src/payloads/services/index.ts [hunk 2]
```diff
@@ -74,11 +78,41 @@ export const getServicesTuple = (
...
+              h()
+            )
+          )
+          .getOrElse(false);
+        return [
+          s.service_id,
+          {
+            is_inbox_enabled: hasSpecialServiceInbox,
+            is_email_enabled: hasSpecialServiceInbox
+              ? faker.datatype.boolean()
```
Comment (Undermaken) [inline src/payloads/services/index.ts:107]:
```suggestion
              ? getRandomValue(false, faker.datatype.boolean(), "services")
```
we are trying to dominate the randomness in the dev-server with the configuration
A dev could go crazy if a value change on every request :dizzy_face:

File: src/routers/features/cgn/index.ts [hunk 3]
```diff
@@ -91,6 +98,22 @@ addHandler(cgnRouter, "get", addPrefix("/activation"), (_, res) =>
...
       };
+
+      const currentPreference = servicesPreferences.get(
+        cgnServiceId as ServiceId
+      );
+
+      const increasedSettingsVersion = currentPreference
+        ? (((currentPreference.settings_version as number) +
+            1) as ServicePreference["settings_version"])
+        : (0 as ServicePreference["settings_version"]);
```
Comment (Undermaken) [inline src/routers/features/cgn/index.ts:109]:
```suggestion
      const increasedSettingsVersion = ((currentPreference?.settings_version ??
        -1) + 1) as NonNegativeInteger;
```
the same, shorter but less readable. On your flavor

File: src/routers/features/cgn/index.ts [hunk 4]
```diff
@@ -91,6 +98,22 @@ addHandler(cgnRouter, "get", addPrefix("/activation"), (_, res) =>
...
+      servicesPreferences.set(cgnServiceId as ServiceId, {
+        is_inbox_enabled: true,
+        is_email_enabled: faker.datatype.boolean(),
```
Comment (Undermaken) [inline src/routers/features/cgn/index.ts:112]:
```suggestion
        is_email_enabled:currentPreference?.is_email_enabled ??
          getRandomValue(false, faker.datatype.boolean(), "services"),
```
the same of the other part: we should avoid the dev madness with a value that could change on every API invocation

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: src/payloads/services/index.ts [hunk 5]
```diff
@@ -74,11 +80,47 @@ export const getServicesTuple = (
...
+
+        return [
+          s.service_id,
+          {
+            is_inbox_enabled: hasSpecialServiceInbox,
+            is_email_enabled: hasSpecialServiceInbox
+              ? getRandomValue(false, faker.datatype.boolean(), "services")
+              : false,
+            is_webhook_enabled: hasSpecialServiceInbox
+              ? faker.datatype.boolean()
```
Comment (Undermaken) [inline src/payloads/services/index.ts:118]:
```suggestion
              ? getRandomValue(false, faker.datatype.boolean(), "services")
```

**Review by Undermaken — CHANGES_REQUESTED** [review]

**Review by Undermaken — APPROVED** [review]

**Pull request merged**